### ISSUE TITLE
[1.9.0-rc4] Containers failing randomly with `[8] System error: invalid argument`

### ISSUE THREAD

**Issue by imikushin** [docker/docker#17577]
The bug looks similar to #17527 and, in fact, is the new behaviour after #17527 has been patched, but sure is not its duplicate. 

I tried to narrow it down a little and here's what I've found. There are a few potential places in the code where this error could be thrown: usages of `libcontainer.SystemError`, all of them are in `vendor/src/github.com/opencontainers/runc/libcontainer/factory_linux.go`. The error is triggered by trying to run concurrent `--net=host` containers. If you run the repro script w/o `--net=host`, the issue goes away. 

Repro: the easiest is to run the following in the docker development container (the one built by `make binary`): 
```
# for i in {1..20}; do (docker run --net=host tianon/true&);done
```
which gives:
```
invalid argument
invalid argument
Error response from daemon: Cannot start container b3371732a3620966ace7d003a3f5b41e51ab110547221c0e211427079047de57: [8] System error: invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
Error response from daemon: Cannot start container 1f0e719676d4f9419c672e46d00eb36e4398862f32c38f256d153a89676d904f: [8] System error: invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
Error response from daemon: Cannot start container cbf72cf664dc100556516c137dd9a41b6061d1aeb76d894e835a9e1149a06055: [8] System error: invalid argument
Error response from daemon: Cannot start container a240f40aaa63963a6552c9db44a8c1534e683f33c569cb602ac3abd06175d701: [8] System error: invalid argument
Error response from daemon: Cannot start container d046cdba9fb2534b5e2712464b11d118c02c1bad3183251c4bb64a1f940eb273: [8] System error: invalid argument
Error response from daemon: Cannot start container 6f881fb3ad7c255e6984dc8eee4628554e0ff03f03e40d3e30b36ab27e4eac33: [8] System error: invalid argument
Error response from daemon: Cannot start container 7beac4495d223d053ebf29a83ad6cc1bf3a7130877c65837b57d0e389fb9f36f: [8] System error: invalid argument
Error response from daemon: Cannot start container 59070ed1ac03aa75a7ec92e592bf3f3fc653a6c80f5ab35e9c352c7b53151b62: [8] System error: invalid argument
Error response from daemon: Cannot start container 5809dd1a6dad4516ae2d73156dd8ce263bc9b31181c33d6c0e026c04f233e47c: [8] System error: invalid argument
Error response from daemon: Cannot start container 3c2a53c8c33aec244de66f858758ad35feb433f17c9dc26f1060ddf602c9d92c: [8] System error: invalid argument
Error response from daemon: Cannot start container 39b58318725fd9135438cd1241f01638e33a42a83b99c6533cc78e7aa5a57330: [8] System error: invalid argument
Error response from daemon: Cannot start container d4a5f1326cf3ac67372a7f045a1c8c048ea44cc15de61817d91504fbe9012d37: [8] System error: invalid argument
Error response from daemon: Cannot start container f6ed91e0926a292b94c309c471347b7998e07c45ae141b05f62dc5d173f66851: [8] System error: invalid argument
Error response from daemon: Cannot start container d3e18e82d422ce52c237f498fb27a06de581a462f79a104024eb0102f5ac2d5c: [8] System error: invalid argument
```

Supplementary info:
`docker version`:
```
Client:
 Version:      1.9.0-rc4
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   e6f5a3c
 Built:        Sat Oct 31 18:47:36 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.0-rc4
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   e6f5a3c
 Built:        Sat Oct 31 18:47:36 UTC 2015
 OS/Arch:      linux/amd64
```

`docker info`:
```
# docker info
Containers: 20
Images: 2
Server Version: 1.9.0-rc4
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 42
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-65-generic
Operating System: Ubuntu 14.04.3 LTS (containerized)
CPUs: 8
Total Memory: 15.63 GiB
Name: a1248732426a
ID: WZH3:IVXS:44HE:THGF:7JRD:PYE3:KUAK:MDZ2:U727:KLJR:WBDW:YBBZ
Debug mode (server): true
 File Descriptors: 12
 Goroutines: 19
 System Time: 2015-10-31T19:30:50.035193143Z
 EventsListeners: 0
 Init SHA1: 
 Init Path: /usr/bin/docker
 Docker Root Dir: /var/lib/docker
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

`uname -a`:
```
Linux a1248732426a 3.13.0-65-generic #105-Ubuntu SMP Mon Sep 21 18:50:58 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```
/cc @ibuildthecloud @tiborvass

**Comment by ibuildthecloud** [comment]
I can confirm that I see this on master in `make shell`.

**Comment by LK4D4** [comment]
Let's close this after merge in docker and confirmation from rancheros.

**Comment by mavenugo** [comment]
yep. I didnt know that the issue can be closed cross projects. with the `fixes` attribute.

**Comment by ibuildthecloud** [comment]
@LK4D4 @mavenugo This patch has fixed this issue and RancherOS boots properly.  On subsequent boot though we are now getting errors such as

```
ERRO[0001] Failed Starting acpid : 500 Internal Server Error: Cannot start container fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713: open : no such file or directory
 
ERRO[0001] Failed Starting syslog : 500 Internal Server Error: Cannot start container d09a5ad8365fed51db83d7e7f231a75f2ca445e9ac4dda704bed80081fbef113: open : no such file or directory
 
ERRO[0001] Failed Starting udev : 500 Internal Server Error: Cannot start container 10eb4c848bf5831b564f2a85d0cec23c658b6d92e18d6133931a100ceb91de26: open : no such file or directory
 
```

**Comment by ibuildthecloud** [comment]
I'll keep investigating

**Comment by ibuildthecloud** [comment]
A bit more info, just grepping for the one of the failed container ids yields

```
time="2015-11-02T03:06:42.679393473Z" level=error msg="failed to delete sandbox c3515b322082ea31ce8d56034fdead0d62d2ad4f0ac3c758ecea978eb9e11a44 while trying to cleanup: could not cleanup all the endpoints in container fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713 / sandbox c3515b322082ea31ce8d56034fdead0d62d2ad4f0ac3c758ecea978eb9e11a44" 
time="2015-11-02T03:06:43.467397721Z" level=info msg="GET /v1.15/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/json" 
time="2015-11-02T03:06:43.492610469Z" level=info msg="GET /v1.15/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/json" 
time="2015-11-02T03:06:43.515437674Z" level=info msg="GET /v1.15/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/json" 
time="2015-11-02T03:06:43.634594900Z" level=info msg="GET /v1.15/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/json" 
time="2015-11-02T03:06:43.637927030Z" level=info msg="POST /v1.15/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/start" 
time="2015-11-02T03:06:43.638719515Z" level=error msg="failed to cleanup up stale network sandbox for container fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713" 
time="2015-11-02T03:06:43.639302203Z" level=error msg="failed to umount /var/lib/system-docker/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/shm: invalid argument" 
time="2015-11-02T03:06:43.639366285Z" level=error msg="failed to umount /var/lib/system-docker/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/mqueue: invalid argument" 
time="2015-11-02T03:06:43.639380499Z" level=error msg="fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713: Failed to umount ipc filesystems: failed to cleanup ipc mounts:\ninvalid argument\ninvalid argument" 
time="2015-11-02T03:06:43.647720373Z" level=warning msg="HostsPath set to \"/var/lib/system-docker/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/hosts\", but can't stat this filename (err = stat /var/lib/system-docker/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/hosts: no such file or directory); skipping" 
time="2015-11-02T03:06:43.647782815Z" level=error msg="Handler for POST /v1.15/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/start returned error: Cannot start container fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713: open : no such file or directory" 
time="2015-11-02T03:06:43.647808677Z" level=error msg="HTTP Error" err="Cannot start container fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713: open : no such file or directory" statusCode=500 
```

**Comment by ibuildthecloud** [comment]
This seems to be related to not having a graceful shutdown in Docker.  In rancheros we develop in KVM and as such most of the time when I reboot I just kill the VM.  Quite rarely do I shutdown gracefully.  If I do gracefully shutdown I don't see this behavior.  I will try to create a simple tests to reproduce.

**Comment by ibuildthecloud** [comment]
@mavenugo Here's a simple way to reproduce.  I run this in the root of the docker git repo.

```bash
docker run --rm --privileged -v $(pwd)/test:/var/lib/docker -v $(pwd)/bundles/latest/binary/docker:/usr/bin/docker -v /etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt -i busybox:buildroot-2014.02 << EOF
docker daemon -D -s vfs &

sleep 1
docker run --name=test --net=host -d -i busybox cat 
killall -9 docker
EOF

docker run --rm --privileged -v $(pwd)/test:/var/lib/docker -v $(pwd)/bundles/latest/binary/docker:/usr/bin/docker -v /etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt -i busybox:buildroot-2014.02 << EOF
docker daemon -D -s vfs &
sleep 1
docker start test
EOF
```

**Comment by mavenugo** [comment]
@ibuildthecloud it seems you are trying to run multiple docker daemons with the same docker root (`-g / --graph`). The sandbox cleanup failures I think are expected in that case. Those are error messages and don't stop the containers from booting. 

But the message -
`time="2015-11-02T03:06:43.647720373Z" level=warning msg="HostsPath set to \"/var/lib/system-docker/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/hosts\", but can't stat this filename (err = stat /var/lib/system-docker/containers/fbf9d23e25f25a54e40273cf09eaa61fc0942a74213b260cc9e15eeb7feed713/hosts: no such file or directory); skipping" ` indicates the missing `/etc/hosts` for the container, which indicates that multiple docker daemon sharing the same docker root is causing this.

Can you please try the scenario with executing each docker daemon's with its dedicated `-g` option ?

@mrjana will have more information to share regarding this.

**Comment by mrjana** [comment]
@mavenugo I don't think @ibuildthecloud is using the same docker root. In fact he is bind mounting $(pwd)/test as his /var/lib/docker for the inside docker. So outside docker and inside docker are in fact running in different docker root.

@ibuildthecloud I was able to repro the problem using your script. Will send out an analysis after investigation.

**Comment by ibuildthecloud** [comment]
@mrjana Thanks.

The test case launches a daemon, kills it, and then starts it again.  I use the same graph dir to simulated a hard reboot, but at no time are two daemons running using the same graph root.

**Comment by mrjana** [comment]
@ibuildthecloud Fixed the problem with your ungraceful restart scenario. Will add the fix as part of the vendoring in libnetwork.

**Comment by ibuildthecloud** [comment]
@mrjana Thanks!  I'll test early tomorrow morning.

**Comment by ibuildthecloud** [comment]
With https://github.com/docker/libnetwork/pull/729 my above test case is not passing, but failing in a different way now.  Will investigate.

**Comment by ibuildthecloud** [comment]
My before mentioned test case fails with
```
time="2015-11-02T15:42:10.885915836Z" level=warning msg="running inside docker container, ignoring missing kernel params: open /proc/sys/net/bridge/bridge-nf-call-iptables: no such file or directory" 
time="2015-11-02T15:42:10.930304569Z" level=debug msg="Assigning addresses for endpoint gateway_4cd4981d7f9e's interface on network docker_gwbridge" 
time="2015-11-02T15:42:10.930410749Z" level=debug msg="RequestAddress(LocalDefault/172.19.0.0/16, <nil>, map[])" 
time="2015-11-02T15:42:10.969015612Z" level=warning msg="Failed detaching sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace from endpoint 0bb87fbb2d6062b03a01bb1f51b7cf41e5d6218df4c101ccb459e8d7b6565850: failed to get network from store during leave: network e38452b5eaae8fb11832953e0fbb291515c90e327e20c656b99ede478463c99d not found\n" 
time="2015-11-02T15:42:10.969309005Z" level=warning msg="Failed deleting endpoint 0bb87fbb2d6062b03a01bb1f51b7cf41e5d6218df4c101ccb459e8d7b6565850: failed to get network during Delete: network e38452b5eaae8fb11832953e0fbb291515c90e327e20c656b99ede478463c99d not found\n" 
time="2015-11-02T15:42:10.969797058Z" level=warning msg="Failed detaching sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace from endpoint 409b33f0c7b0d3857c913f4ecc53c9fd09b10e232fc1269ba24978cf459906ea: container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: endpoint create on GW Network failed: service endpoint with name gateway_4cd4981d7f9e already exists\n" 
time="2015-11-02T15:42:10.969872152Z" level=error msg="Error deleting sandbox id aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace for container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: could not cleanup all the endpoints in container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543 / sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace" 
time="2015-11-02T15:42:10.969927457Z" level=error msg="failed to umount /var/lib/docker/containers/4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543/shm: invalid argument" 
time="2015-11-02T15:42:10.969957646Z" level=error msg="failed to umount /var/lib/docker/containers/4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543/mqueue: invalid argument" 
time="2015-11-02T15:42:10.969976181Z" level=error msg="4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: Failed to umount ipc filesystems: failed to cleanup ipc mounts:\ninvalid argument\ninvalid argument" 
time="2015-11-02T15:42:10.970083602Z" level=error msg="Handler for POST /v1.22/containers/test/start returned error: Cannot start container test: container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: endpoint join on GW Network failed: failed to add interface veth007a19b to sandbox: error renaming interface \"veth007a19b\" to \"eth0\": file exists" 
time="2015-11-02T15:42:10.970122560Z" level=error msg="HTTP Error" err="Cannot start container test: container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: endpoint join on GW Network failed: failed to add interface veth007a19b to sandbox: error renaming interface \"veth007a19b\" to \"eth0\": file exists" statusCode=500 
Error response from daemon: Cannot start container test: container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: endpoint join on GW Network failed: failed to add interface veth007a19b to sandbox: error renaming interface "veth007a19b" to "eth0": file exists
Error: failed to start containers: [test]
```

**Comment by mavenugo** [comment]
Yes. It is not the complete fix. Pls hold on for libnetwork vendor in

**Comment by mrjana** [comment]
@ibuildthecloud #17594 updated with new libnetwork code. Can you please try it out and let us know if it fixes all of your issues?

**Comment by ibuildthecloud** [comment]
@mavenugo I'll try now, thanks

**Comment by ibuildthecloud** [comment]
@mavenugo Still seeing failures with test case

```
time="2015-11-02T17:01:56.804493431Z" level=warning msg="Failed detaching sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace from endpoint 0bb87fbb2d6062b03a01bb1f51b7cf41e5d6218df4c101ccb459e8d7b6565850: failed to get network from store during leave: network e38452b5eaae8fb11832953e0fbb291515c90e327e20c656b99ede478463c99d not found\n" 
time="2015-11-02T17:01:56.804546462Z" level=warning msg="Failed deleting endpoint 0bb87fbb2d6062b03a01bb1f51b7cf41e5d6218df4c101ccb459e8d7b6565850: failed to get network during Delete: network e38452b5eaae8fb11832953e0fbb291515c90e327e20c656b99ede478463c99d not found\n" 
time="2015-11-02T17:01:56.804571315Z" level=warning msg="Failed detaching sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace from endpoint 409b33f0c7b0d3857c913f4ecc53c9fd09b10e232fc1269ba24978cf459906ea: failed to get network from store during leave: network 51c8174676335344a882db75d0ce6ffff0611d5cb336e7fb43881e1076ef49de not found\n" 
time="2015-11-02T17:01:56.804592438Z" level=warning msg="Failed deleting endpoint 409b33f0c7b0d3857c913f4ecc53c9fd09b10e232fc1269ba24978cf459906ea: failed to get network during Delete: network 51c8174676335344a882db75d0ce6ffff0611d5cb336e7fb43881e1076ef49de not found\n" 
time="2015-11-02T17:01:56.804611535Z" level=error msg="failed to cleanup up stale network sandbox for container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543" 
time="2015-11-02T17:01:56.948386831Z" level=warning msg="Failed detaching sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace from endpoint 0bb87fbb2d6062b03a01bb1f51b7cf41e5d6218df4c101ccb459e8d7b6565850: failed to get network from store during leave: network e38452b5eaae8fb11832953e0fbb291515c90e327e20c656b99ede478463c99d not found\n" 
time="2015-11-02T17:01:56.948547070Z" level=warning msg="Failed deleting endpoint 0bb87fbb2d6062b03a01bb1f51b7cf41e5d6218df4c101ccb459e8d7b6565850: failed to get network during Delete: network e38452b5eaae8fb11832953e0fbb291515c90e327e20c656b99ede478463c99d not found\n" 
time="2015-11-02T17:01:56.948602603Z" level=warning msg="Failed detaching sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace from endpoint 409b33f0c7b0d3857c913f4ecc53c9fd09b10e232fc1269ba24978cf459906ea: failed to get network from store during leave: network 51c8174676335344a882db75d0ce6ffff0611d5cb336e7fb43881e1076ef49de not found\n" 
time="2015-11-02T17:01:56.948977291Z" level=warning msg="Failed deleting endpoint 409b33f0c7b0d3857c913f4ecc53c9fd09b10e232fc1269ba24978cf459906ea: failed to get network during Delete: network 51c8174676335344a882db75d0ce6ffff0611d5cb336e7fb43881e1076ef49de not found\n" 
time="2015-11-02T17:01:56.950501227Z" level=warning msg="Failed detaching sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace from endpoint 2cc96377358364f2534d104ab68b7dc1d1a23212ae73caef8b215697149a573f: container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: endpoint create on GW Network failed: service endpoint with name gateway_4cd4981d7f9e already exists\n" 
time="2015-11-02T17:01:56.950890508Z" level=error msg="Error deleting sandbox id aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace for container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: could not cleanup all the endpoints in container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543 / sandbox aa5c4fff6da1e122b6dff6eff737130e0509ecc39753c72c432016e465a65ace" 
time="2015-11-02T17:01:56.950982504Z" level=error msg="failed to umount /var/lib/docker/containers/4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543/shm: invalid argument" 
time="2015-11-02T17:01:56.951092000Z" level=error msg="failed to umount /var/lib/docker/containers/4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543/mqueue: invalid argument" 
time="2015-11-02T17:01:56.951117042Z" level=error msg="4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: Failed to umount ipc filesystems: failed to cleanup ipc mounts:\ninvalid argument\ninvalid argument" 
time="2015-11-02T17:01:56.951283347Z" level=error msg="Handler for POST /v1.22/containers/test/start returned error: Cannot start container test: container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: endpoint create on GW Network failed: service endpoint with name gateway_4cd4981d7f9e already exists" 
time="2015-11-02T17:01:56.951341453Z" level=error msg="HTTP Error" err="Cannot start container test: container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: endpoint create on GW Network failed: service endpoint with name gateway_4cd4981d7f9e already exists" statusCode=500 
Error response from daemon: Cannot start container test: container 4cd4981d7f9e4b5db81554613dd93c5b189ecdbd8d44812148e247a7c966a543: endpoint create on GW Network failed: service endpoint with name gateway_4cd4981d7f9e already exists
Error: failed to start containers: [test]
```

**Comment by mavenugo** [comment]
@ibuildthecloud unfortunately, am unable to try your script (have environment issues).
but i believe you meant to ping @mrjana who got your script working and fixed the issue in #17594.
can you please confirm that you took the latest from #17594 including both the docker side and libnetwork side changes ?

**Comment by mrjana** [comment]
@ibuildthecloud If you are trying to rebuild docket from the same git branch make sure you remove the dind image and rebuild. Sometimes docker tries to rebuild from old cache of docker source and hence does not reflect the latest changes. Need to investigate further and file a separate issue for that. Also remove the "test" directory and rerun your script.

**Comment by ibuildthecloud** [comment]
@mrjana Let last run I didn't delete the ./test folder.  Deleting the test folder it worked.  So I'm moving onto testing in RancherOS

**Comment by kladkogex** [comment]
Why was this closed ? We see it on Ubuntu 15 with the latest docker

**Comment by matevarga** [comment]
Same here, I see this behaviour with 1.10.1. :(

**Comment by jadbox** [comment]
I have the same issue sadly:

```
Starting vast_vtracking_12 ... error

ERROR: for vast_vtracking_12  500 Internal Server Error: container 8125192b69a6006fffa1c349802e631791d1f7be84083a4e45c30e5a2d6f3966: endpoint create on GW Network failed:
 service endpoint with name gateway_8125192b69a6 already exists 
```
